vault/request_handling.go

```diff
@@ -1686,6 +1687,13 @@ func (c *Core) handleLoginRequest(ctx context.Context, req *logical.Request) (re
 		// Attach the display name, might be used by audit backends
 		req.DisplayName = auth.DisplayName
 
+		// If this is not a role-based quota, we still need to associate the
+		// login role with this lease for later lease-count quotas to be
+		// accurate.
+		if reqRole == nil && resp.Auth.TokenType != logical.TokenTypeBatch {
```

Comment:
Is this how we determine if a request generates a lease? just make sure it's not a batch token?

Reply:
Yes, for login requests.  In general it's more complicated.